app/models/profile.rb

```diff
@@ -70,5 +70,22 @@ def from_openscap_parser(op_profile, benchmark_id: nil, account_id: nil)
 
       profile
     end
+
+    def rules_and_rule_groups(profile)
+      profile_rules = profile.rules
+      profile_rule_groups = profile.rule_groups
+      custom_arrange(profile_rule_groups, profile_rules)
+    end
+
+    def custom_arrange(rule_groups, rules)
+      arranged_rule_groups = rule_groups.includes(:rules).arrange_serializable do |parent, children|
+        rule_children = parent.rules.map do |r|
+          { rule: r.id }
+        end
+        { rule_group: parent.id, group_children: children, rule_children: rule_children }
+      end
+
+      arranged_rule_groups.concat(rules.without_rule_group_parent.map { |pr| { rule: pr.id } }).to_json
+    end
```

Comment:
@skateman @romanblanco After working on the fix for the profile_rule_groups table and understanding the relationship between the profile, rules and groups more I was able to simplify the ruby solution a good amount. 

This takes about .067s so it's pretty fast and I was able to remove a lot of the maps. I checked the logs and this only runs 5 queries total so a query isn't being ran inside the loop anymore. The unfinished recursive PostgreSQL solution is already around 0.262s and that's only setting up the rule groups in a hierarchical format, not factoring rules in yet or converting to json.

 I'll do some more testing after the [profile_rule_groups table fix](https://github.com/RedHatInsights/compliance-backend/pull/1476) is merged but let me know what you guys think about this!